sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRegistryClientLiveTests.cs

```diff
@@ -42,7 +43,62 @@ await foreach (string repository in repositories)
                 }
             }
 
+            // Assert
             Assert.IsTrue(gotHelloWorld);
         }
+
+        [RecordedTest]
+        public async Task CanGetRepositoriesWithCustomPageSize()
+        {
+            // Arrange
+            var client = CreateClient();
+            int pageSize = 2;
+            int minExpectedPages = 2;
+
+            // Act
+            AsyncPageable<string> repositories = client.GetRepositoriesAsync();
+            var pages = repositories.AsPages(pageSizeHint: pageSize);
+
+            int pageCount = 0;
+            await foreach (var page in pages)
+            {
+                Assert.IsTrue(page.Values.Count <= pageSize);
+                pageCount++;
+            }
+
+            // Assert
+            Assert.IsTrue(pageCount >= minExpectedPages);
+        }
+
+        [RecordedTest]
+        public async Task CanStartPagingMidCollection()
+        {
+            // Arrange
+            var client = CreateClient();
+            int pageSize = 1;
+            int minExpectedPages = 2;
+
+            // Act
+            AsyncPageable<string> repositories = client.GetRepositoriesAsync();
+            var pages = repositories.AsPages($"</acr/v1/_catalog?last=library/alpine&n={pageSize}>");
```

Comment:
Hmm, doesn't pageSize come in through the second parameter of AsPages? This goes back to my other comment about the `n` param not being respected. Would AsPages only work with this overload where you specify the pagesize as a query param, as opposed to the other overload that has a separate pageSizeHint param?

Reply:
Yeah, this isn't ideal.  I chatted with @pakrym about this, and he helped me understand that AsyncPageable.AsPages() is considered an advanced method.  If we learn it is a common scenario that people need to start in the middle of a collection, we'll simplify this API, probably by adding a parameter to the GetRepositories() method.